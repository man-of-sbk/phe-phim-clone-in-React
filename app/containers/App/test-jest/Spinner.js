import React from 'react';
import PropTypes from 'prop-types';

const Spinner = ({ loading, size, color }) =>
  loading ? (
    <div className={`icon icon-spin text-center ${size} ${color}`} />
  ) : null;

Spinner.propTypes = {
  loading: PropTypes.bool,
  size: PropTypes.number,
  color: PropTypes.string,
};

export default Spinner;
