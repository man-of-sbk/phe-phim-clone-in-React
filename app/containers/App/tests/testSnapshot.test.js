import React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'react-testing-library';
import Spinner from '../test-jest/Spinner';

// *** basic snapshot testing
(() => {
  // it('renders correctly', () => {
  //   const props = {
  //     loading: true,
  //     size: 60,
  //     color: 'blue',
  //   };
  //   expect(props).toMatchSnapshot();
  // });
})();

// *** snapshot with raw react component using react-test-renderer
(() => {
  it('renders correctly', () => {
    const props = {
      loading: true,
      size: 60,
      color: 'blue',
    };
    const tree = renderer.create(<Spinner {...props} />).toJSON();

    console.log(typeof tree);
    console.log(<Spinner {...props} />);

    expect(tree).toMatchSnapshot();
  });
})();

// *** snapshot with raw react component using react-testing-library
(() => {
  // it('renders correctly', () => {
  //   const props = {
  //     loading: true,
  //     size: 60,
  //     color: 'blue',
  //   };
  //   const tree = render(<Spinner {...props} />);
  //   console.log(tree);
  //   expect(tree).toMatchSnapshot();
  // });
})();
