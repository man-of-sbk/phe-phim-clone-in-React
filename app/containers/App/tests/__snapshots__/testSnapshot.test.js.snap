// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="icon icon-spin text-center 60 blue"
/>
`;
